src/XR/features/WebXRMeshDetector.ts

```diff
@@ -0,0 +1,238 @@
+     */
+    doNotRemoveMeshesOnSessionEnded?: boolean;
+    /**
+     * Preferred detector configuration, not all preferred options will be supported by all platforms.
+     */
+    preferredDetectorOptions?: XRGeometryDetectorOptions;
+}
+
+/**
+ * A babylon interface for a XR mesh.
+ *
+ * Currently not supported by WebXR, available only with BabylonNative
+ */
+export interface IWebXRMesh {
+    /**
+     * a babylon-assigned ID for this mesh
+     */
+    id: number;
+    /**
+     * an array of vertex positions in babylon space. right/left hand system is taken into account.
+     */
+    positions: Float32Array;
+    /**
+     * an array of indices in babylon space. right/left hand system is taken into account.
+     */
+    indices: Uint32Array;
+    /**
+     * an array of vertex normals in babylon space. right/left hand system is taken into account.
+     */
+    normals?: Float32Array;
+    /**
+     * data required for construction a mesh in Babylon.js
+     */
+    xrMesh: XRMesh;
+    /**
+     * A transformation matrix to apply on the mesh that will be built using the meshDefinition
+     * Local vs. World are decided if worldParentNode was provided or not in the options when constructing the module
+     */
+    transformationMatrix: Matrix;
+}
+
+let meshIdProvider = 0;
+
+/**
+ * The mesh detector is used to detect meshes in the real world when in AR
+ */
+export class WebXRMeshDetector extends WebXRAbstractFeature {
+    private _detectedMeshes: Array<IWebXRMesh> = [];
+    private _lastDetectedSet: XRMeshSet = new Set();
+
+    /**
+     * The module's name
+     */
+    public static readonly Name = WebXRFeatureName.MESH_DETECTION;
+    /**
+     * The (Babylon) version of this module.
+     * This is an integer representing the implementation version.
+     * This number does not correspond to the WebXR specs version
+     */
+    public static readonly Version = 1;
+
+    /**
+     * Observers registered here will be executed when a new mesh was added to the session
+     */
+    public onMeshAddedObservable: Observable<IWebXRMesh> = new Observable();
+    /**
+     * Observers registered here will be executed when a mesh is no longer detected in the session
+     */
+    public onMeshRemovedObservable: Observable<IWebXRMesh> = new Observable();
+    /**
+     * Observers registered here will be executed when an existing mesh updates
+     */
+    public onMeshUpdatedObservable: Observable<IWebXRMesh> = new Observable();
+
+    constructor(_xrSessionManager: WebXRSessionManager, private _options: IWebXRMeshDetectorOptions = {}) {
+        super(_xrSessionManager);
+        this.xrNativeFeatureName = "mesh-detection";
+        this._initMeshDetector = this._initMeshDetector.bind(this);
+        if (this._xrSessionManager.session) {
+            this._initMeshDetector(this._xrSessionManager.session);
+        } else {
+            this._xrSessionManager.onXRSessionInit.addOnce(this._initMeshDetector);

```

Comment:
same issue as mentioned before with context.

Reply:
This should now be reverted